### ISSUE TITLE
When find_files is open via the insert mode shortcut the 'Buffers' field is prepopulated with 'A'

### ISSUE THREAD

**Issue by Glavnokoman** [nvim-telescope/telescope.nvim#1931]
### Description

Subj. The same happens in several other Telescope functions like 'buffers' for example. This only started to happen recently: a week or two.

### Neovim version

```markdown
NVIM v0.7.0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

Features: +acl +iconv +tui
```

### Operating system and version

Linux 5.13.0-39 x86_64 GNU/Linux

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 11.0.2
  - OK: fd: found fd 7.4.0

## ===== Installed extensions =====
```

### Steps to reproduce

switch to insert mode
press `<c-w>o`

### Expected behavior

empty 'buffer' field

### Actual behavior

'buffer' field prepopulated with 'A'

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
vim.cmd [[inoremap <c-w><c-o> <c-o><cmd>Telescope buffers<cr>]]
vim.cmd [[inoremap <c-w>o     <c-o><cmd>Telescope find_files<cr>]]

local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by fdschmidt93** [comment]
> inoremap <c-w>o Telescope find_files
switch to insert mode
press <c-w>o

Just to make sure, the mapping should be `inoremap <c-w>o <cmd>Telescope find_files<CR>`.

With that, I cannot reproduce with `minimal_init.lua` on both neovim `master` and `stable`. Either the way you map messes with mode setting at picker startup, or do you have any autocommands etc in your config that may affect mode switching in way?

In other words, please actually provide a proper `minimal_init.lua` or more info for reproduction steps.

**Comment by Glavnokoman** [comment]
- yes, the mapping is with `<cmd>` I did not notice formatting has eaten that.
- I am terribly sorry, it is indeed not reproducible with the minimal config, any ideas how do I debug it further? I do not have (TBMK) any auto-commands messing with the mode switch...

**Comment by Glavnokoman** [comment]
Ha! I moved the bindings to the minimal config and now it is reproducible! Please take a look. And sorry again for not checking it on submission.

**Comment by Glavnokoman** [comment]
Thanks! That solves it for me. It works indeed without the `<c-o>`. 
However it used to work some time ago, so may still be an indicator of some hidden bug. Up to you :).

**Comment by fdschmidt93** [comment]
> However it used to work some time ago, so may still be an indicator of some hidden bug. Up to you :).

Ah, no worries, it's not (really). Initial mode setting is extremely tricky business and we only recently got it "right" (for I'm reasonably confident 99% of usage). "Right" of course in quotation marks as you've proven there's yet another edge case, though thankfully this one trivially solved once fully understood where it stems from :)

This one we, assuming I understand correctly, could not address in principled fashion anyways since mode setting is hidden from us as to how `\<c-o>` works from within insert mode. Consequently, telescope mode setting logic gets confused, since we account for launching pickers from insert mode, though in your case to telescope it appears as if you are triggering from normal mode while actually being in insert mode.

Glad, we've got this sorted, so closing :)
